### PR TITLE
chore: upgrade dependencies

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Our version of Kotlin (1.6.21) is a bit behind the newest stable version (1.7.10)

### Solutions

Update it, and all the needed paraphernalia around it.

> **Note**: Some dependencies needed upgrade to be compatible with Kotlin 1.7.10.
> **Note**: I've ran code formatting in `WelcomeScreen` by accident after correcting some imports, but I think it's not a bad thing to do it in this tiny PR.

### Dependencies

Needs releases with:

- https://github.com/wireapp/kalium/pull/779

### Testing

Ran the app manually, everything seems to be working nice.

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/2/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4129135)(4129135)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4129135)(4129135)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4129330)(4129330)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4129330)(4129330)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4130076)(4130076)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4130076)(4130076)-dev-release.apk)

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -51,28 +51,28 @@ object Repositories {
...
-        const val composeMaterial = compose
+        const val composeMaterial = "1.2.1"
         const val composeMaterial3 = "1.0.0-alpha11"
         const val composeActivity = "1.4.0"
-        const val composeNavigation = "2.4.2"
+        const val composeNavigation = "2.5.0"
         const val accompanist = "0.24.7-alpha" // https://github.com/google/accompanist/issues/1162
         const val composeConstraint = "1.0.0"
-        const val hilt = "2.38.1"
+        const val hilt = "2.43.2"
```
Comment (ohassine) [inline buildSrc/src/main/kotlin/Dependencies.kt:72]: Maybe we need also to upgrade hilt plugin version ? ( `com.google.dagger:hilt-android-gradle-plugin` inside `build.gradle.kts` in `buildSrc` module)

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 2]
```diff
@@ -51,28 +51,28 @@ object Repositories {
...
-        const val ktx = "1.6.0"
+        const val ktx = "1.8.0"
         const val material = "1.5.0"
         const val pinEditText = "1.2.3"
         const val sqlLiteJdbc = "3.36.0"
         const val desugaring = "1.1.5"
         const val workManager = "2.7.1"
-        const val fragment = "1.2.5"
+        const val fragment = "1.5.1"
         const val compose = "1.2.0-beta01"
```
Comment (MohamadJaara) [inline buildSrc/src/main/kotlin/Dependencies.kt:65]: Compose 1.2.1 is in stable 
```suggestion
        const val compose = "1.2.1"
```

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 3]
```diff
@@ -110,7 +110,7 @@ object Libraries {
     const val accompanistNavAnimation   = "com.google.accompanist:accompanist-navigation-animation:${Versions.accompanist}"
     const val accompanistIndicator      = "com.google.accompanist:accompanist-pager-indicators:${Versions.accompanist}"
     const val material                  = "com.google.android.material:material:${Versions.material}"
-    const val visibilityModifiers       = "io.github.esentsov:kotlin-visibility:${Versions.visibilityModifiers}"
+    const val visibilityModifiers        = "io.github.esentsov:kotlin-visibility:${Versions.visibilityModifiers}"
```
Comment (MohamadJaara) [inline buildSrc/src/main/kotlin/Dependencies.kt:113]:
```suggestion
    const val visibilityModifiers       = "io.github.esentsov:kotlin-visibility:${Versions.visibilityModifiers}"
```

**Review by yamilmedina — APPROVED** [review]
Awesome!!!! nice to have these dependencies updated 🥇 👏

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/6/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4132980)(4132980)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4132980)(4132980)-dev-release.apk)

**Review by trOnk12 — APPROVED** [review]

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 3 above)
Comment (vitorhugods) [reply buildSrc/src/main/kotlin/Dependencies.kt:113]: This is how it looks to me now

image

What is this wizzardry?

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 2 above)
Comment (vitorhugods) [reply buildSrc/src/main/kotlin/Dependencies.kt:65]: I raise you `1.3.0`

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 3 above)
Comment (MohamadJaara) [reply buildSrc/src/main/kotlin/Dependencies.kt:113]: image

**Review by ohassine — APPROVED** [review]
🚀🚀

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/8/) **failed**.

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 3 above)
Comment (vitorhugods) [reply buildSrc/src/main/kotlin/Dependencies.kt:113]: Yes, on GitHub it looks weird. On the IDE it looks aligned.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/9/) **failed**.

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/10/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-852/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4135099)(4135099)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4135099)(4135099)-dev-release.apk)

**Pull request merged**